src/lib/containers/access_requirements/ImposeRestrictionDialog.tsx

```diff
@@ -0,0 +1,132 @@
+import React from 'react'
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  FormControl,
+  FormLabel,
+  IconButton,
+  Stack,
+  Typography,
+} from '@mui/material'
+import { RadioGroup } from '../widgets/RadioGroup'
+import FullWidthAlert from '../FullWidthAlert'
+import IconSvg from '../IconSvg'
+import { useCreateLockAccessRequirement } from '../../utils/hooks/SynapseAPI'
+import { displayToast } from '../ToastMessage'
+import { HelpPopover } from '../HelpPopover'
+
+export type ImposeRestrictionFormProps = {
+  /* The ID of the entity that the user may choose to restrict */
+  entityId: string
+  open: boolean
+  onClose: () => void
+}
+
+export default function ImposeRestrictionDialog(
+  props: ImposeRestrictionFormProps,
+) {
+  const NOT_HUMAN_DATA_WARNING = (
+    <Typography variant={'smallText1'}>
+      Sage Bionetworks does not typically impose conditions for use on non-human
+      data unless there is a legal, ethical or regulatory reason to do so. If
+      you want to add conditions for use to this content, please contact the
+      Synapse Access and Compliance Team (ACT) to discuss at{' '}
+      <a href={'mailto:act@sagebase.org'}>act@sagebase.org</a>.
+    </Typography>
+  )
+
+  const { entityId, open, onClose } = props
+  const [isSensitiveHumanData, setIsSensitiveHumanData] = React.useState<
+    boolean | undefined
+  >(undefined)
+
+  const { mutate: createLockedAccessRequirement, isLoading } =
+    useCreateLockAccessRequirement({
+      onSuccess: () => {
+        displayToast('Successfully imposed restriction', 'success')
+        onClose()
+      },
+      onError: e => {
+        displayToast(`Failed to impose restriction: ${e.reason}`, 'danger')
+      },
+    })
+
+  function okClickedHandler() {
+    if (isSensitiveHumanData) {
+      createLockedAccessRequirement(entityId)
+    } else {
+      onClose()
+    }
+  }
+
+  function radioOnChangeHandler(newValue: boolean) {
+    setIsSensitiveHumanData(newValue)
+  }
+
+  return (
+    <Dialog open={open} onClose={onClose} maxWidth={'sm'} fullWidth>
```

Comment:
Prefer MUI over Bootstrap dialog because CSS conflicts between SRC/SWC make this a headache, and it's really challenging to correctly apply the `bootstrap-4-backport` class on the modal, since it is rendered in a distinct subtree of the DOM that we can't wrap